### PR TITLE
Added the ability to set custom actions on an alert.

### PR DESCRIPTION
**Pull request by ExploreMqt**

### CODE REVIEW

File: lib/apnagent/message.js [hunk 1]
```diff
@@ -139,12 +139,18 @@ Message.prototype.alert = function (key, value) {
...
     , 'action-loc-key'
+    , 'actions'
     , 'loc-key'
     , 'loc-args'
     , 'launch-image'
     , 'title'
     , 'title-loc-key'
     , 'title-loc-args'
+ ],
+      valueMustBeArray = [
```
Comment (logicalparadox) [inline lib/apnagent/message.js]: style ocd: of your `var` declaration is more than one line then use the `var` keyword for each declaration....

```js
var arr1 = [
  // ..
]

var arr2 = [
  // ...
]
```

File: lib/apnagent/message.js [hunk 2]
```diff
@@ -155,6 +161,7 @@ Message.prototype.alert = function (key, value) {
     this.aps['body'] = key;
   } else {
     if (!~allowed.indexOf(key)) return this;
+    if (~valueMustBeArray.indexOf(key) && value.constructor != Array) return this;
```
Comment (logicalparadox) [inline lib/apnagent/message.js]: You can use `Array.isArray(value)` or https://github.com/chaijs/type-detect

**Comment by ExploreMqt** [comment]
Good call on the use of Array.isArray() thanks. Still learning JS.

**Comment by ExploreMqt** [comment]
The failure is totally unrelated to the changes made. I have however seen that same test fail one time locally. I've looked at the test and I'm not sure why it behaves unpredictably. Thoughts?

**Comment by logicalparadox** [comment]
`Cache#push` has an optional third argument for age, which is a timestamp. Looks like the test that is failing is not including the `now` variable, so the timestamp doesn't match if it took more than 1ms to push all four entries into the cache.

**Comment by logicalparadox** [comment]
LGTM: ready to merge?

**Comment by ExploreMqt** [comment]
I think so.

**Pull request merged**